packages/api/db/migration/20240125135642_add_user_last_name_default.js

```diff
@@ -0,0 +1,26 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.alterTable('users', (table) => {
+    table.string('last_name').notNullable().defaultTo('').alter();
```

Comment:
What existed before was really weird... `notNullable().defaultTo(null)` 😂 .  Almost makes me think it was a typo.

My question is why default to empty string instead of just allowing null with `.nullable()`? It might make it easier to deal with on the frontend but it seems weird to me https://stackoverflow.com/a/44890240

Reply:
Well the not nullable with a null default is not that weird, essentially null is the default if there's no default if that makes sense. Essentially if you set a column to not nullable and you don't set a default, you're expecting that whoever's inserting a new record will always specify that field, which is totally fine if the field is required and that's what you want to enforce.

To the second point, I'd totally prefer to make this nullable, but that seems higher risk especially in the context of a hotfix since there might be places where the client is expecting a string and is not prepared with nullish coalescing to handle the null scenario, so this seems like the safest choice.

Reply:
Ok I did not realize it was hotfix 👍